config/manager/deployment.yaml

```diff
@@ -31,6 +31,12 @@ spec:
         securityContext:
           allowPrivilegeEscalation: false
           readOnlyRootFilesystem: true
+          capabilities:
+            drop: [ "ALL" ]
+          runAsUser: 65534
```

Comment:
Setting a used ID here was proposed before, but we’ve rejected it due to breaking Flux on OpenShift and other Kubernetes distributions that enforce a specific ID range. I think this change should be made to the docs here https://fluxcd.io/docs/installation/#pod-security-policy

Reply:
Think that shipping a default configuration and instructing people to change this if it does not work for their distribution is better than hoping people will find your security notes (and then _also_ apply the required configuration).

Reply:
@stefanprodan when this was last attempted, do you remember what user/group id was used?

The user ID picked here is linked to the user `nobody`, which is meant to be the user with least privileges and is generally defined as part of all main distributions.

The upstream [pause container](https://github.com/kubernetes/kubernetes/blob/master/build/pause/Dockerfile#L19) relies on the same approach since Kubernetes 1.21. Other security focused projects targetting OpenShift also [rely on it](https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/Dockerfile.ubi#L47).

I completely agree with you that we should not prioritise security over reliability. On that point, would this be better received if accompanied with an OpenShift E2E automated test?

Reply:
I think it throws an error like this: `unable to validate against any security context constraint: [fsGroup: Invalid value: []int64{65534}: 65534 is not an allowed group spec.containers[0].securityContext.securityContext.runAsUser: Invalid value: 1001: must be in the ranges: [1000810000, 1000819999]]`

Reply:
We do have e2e tests for OpenShift, Flux is on OperatorHub. Adding fs group or user ID to each controller repo, just to remove them in flux CLI makes zero sense to me.

Reply:
Why would we remove them again in the Flux CLI if the OpenShift / OperatorHub setup has custom patches to make it work anyway?

Reply:
They get removed [here](https://github.com/weaveworks/flux2-openshift/blob/main/kustomization.yaml#L16), not in CLI my bad.